am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -263,6 +246,51 @@ public FilteredResourceEnvelope filterResource(@NotBlank String userId,
         return permissionsService.merge(permissionsForRoles);
     }
 
+    /**
+     * Retrieves {@link RolePermissions} filtered by role security classification for a specific resource and role.
+     *
+     * @param resourceDefinition {@link ResourceDefinition} a unique service name, resource type and resource name
+     * @param roleName           user role name
+     * @return {@link RolePermissions} a combination of permissions and security classifications for a role name
+     * @throws PersistenceException if any persistence errors were encountered
+     */
+    @AuditLog("returned role access to resource defined as '{{resourceDefinition.serviceName}}|"
+        + "{{resourceDefinition.resourceType}}|{{resourceDefinition.resourceName}}' for role '{{roleName}}': "
+        + "{{result}}")
+    public RolePermissions getRolePermissions(@NotNull @Valid ResourceDefinition resourceDefinition,
```

Comment:
I'd consider the following steps:

1. get access type and security classification for role (*)
2. calculate visible security classifications using utility
3. fetch from DB all attributes that are visible for role alongside their security classifications and permissions (*)

(*) is used to mark DB calls

Obviously there is no data structure for an output needed in step 3 but so called query class could be created in designated package.